neo/api/JSONRPC/test_json_rpc_api.py

```diff
@@ -516,3 +514,34 @@ def test_listaddress_with_wallet(self):
         self.app.wallet.Close()
         self.app.wallet = None
         os.remove(test_wallet_path)
+
+    def test_getnewaddress_no_wallet(self):
+        req = self._gen_rpc_req("getnewaddress", params=[])
+        mock_req = mock_request(json.dumps(req).encode("utf-8"))
+        res = json.loads(self.app.home(mock_req))
+
+        error = res.get('error', {})
+
+        self.assertEqual(error.get('code', None), -400)
+        self.assertEqual(error.get('message', None), "Access denied.")
+
+    def test_getnewaddress_with_wallet(self):
+        test_wallet_path = "./listaddress.db3"
```

Comment:
i think a slightly better approach would be using the Tempfile module to create a temp wallet file in a standard temp directory: https://docs.python.org/3/library/tempfile.html